src/features/bookmarkSlice.js

```diff
@@ -0,0 +1,111 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+import {
+  bookmarkService,
+  removeFromBookmarkService,
+} from "../services/bookmarkServices";
+
+const initialState = {
+  isLoading: false,
+  bookmarkPosts: [],
+  error: "",
+};
+
+export const getBookMarkPosts = createAsyncThunk(
+  "bookmark/getBookMarkPosts",
+  async ({ token }, { rejectWithValue }) => {
+    try {
+      const res = await bookmarkService(token);
+      return res.data;
+    } catch (error) {
+      return rejectWithValue({ message: "Getting bookmarks failed" });
+    }
+  }
+);
+
+export const bookmarkPost = createAsyncThunk(
+  "bookmark/bookmarkPost",
+  async ({ postId, token }, { rejectWithValue }) => {
+    try {
+      const res = await bookmarkService(postId, token);
+      if (res.status === 200) return res.data;
+      return res.data;
+    } catch (error) {
+      return rejectWithValue({ message: "Bookmark post failed" });
+    }
+  }
+);
+
+export const removeFromBookmark = createAsyncThunk(
+  "bookmark/removeFromBookmark",
+  async ({ postId, token }, { rejectWithValue }) => {
+    try {
+      const res = await removeFromBookmarkService(postId, token);
+      if (res.status === 200) return res.data;
+    } catch (error) {
+      return rejectWithValue({ message: "Rmove post from bookmark failed" });
+    }
+  }
+);
+
+const bookmarkSlice = createSlice({
+  name: "bookmark",
+  initialState,
+  extraReducers: (builder) => {
```

Comment:
you can also do this, extraReducers: [getBookMarkPosts.pending] : (state, {payload}) => {}[getBookMarkPosts.rejected] : (state, {payload}) => {}

doesnt make a difference tho

Reply:
> you can also do this, extraReducers: [getBookMarkPosts.pending] : (state, {payload}) => {}[getBookMarkPosts.rejected] : (state, {payload}) => {}
> 
> doesnt make a difference tho

ok, I will remove it in next PR.